Docker Build Fails on Latest Main Branch

# Docker Build Fails on Latest Main Branch

## Description
When cloning the repository and running `docker compose up --build` on the latest `main` branch, the build fails with an error. However, the build succeeds when checking out to commit `6bdf67708238757542cb5e2094c50fa4e2cacf44`.

## Steps to Reproduce
1. Clone the repository
2. Checkout the latest `main` branch
3. Run `docker compose up --build`

## Error Message
```
WARN[0000] /Users/efexp/Desktop/Source/apache-iceberg-101-ja/docker-compose.yml: `version` is obsolete 
[+] Building 2.0s (12/14)                                                                                           docker:orbstack
 => [rest internal] load build definition from Dockerfile                                                                      0.0s
 => => transferring dockerfile: 831B                                                                                           0.0s
 => [rest internal] load metadata for docker.io/azul/zulu-openjdk:17                                                           1.3s
 => [rest internal] load metadata for docker.io/azul/zulu-openjdk:17-jre-headless                                              1.3s
 => [rest auth] azul/zulu-openjdk:pull token for registry-1.docker.io                                                          0.0s
 => [rest internal] load .dockerignore                                                                                         0.0s
 => => transferring context: 83B                                                                                               0.0s
 => [rest internal] load build context                                                                                         0.0s
 => => transferring context: 124.56kB                                                                                          0.0s
 => [rest stage-1 1/4] FROM docker.io/azul/zulu-openjdk:17-jre-headless@sha256:5aac999062b505b47e50daf1c10a9273e28f45a711ce44  0.0s
 => CACHED [rest stage-1 2/4] RUN     set -xeu &&     groupadd iceberg --gid 1000 &&     useradd iceberg --uid 1000 --gid 100  0.0s
 => CACHED [rest builder 1/4] FROM docker.io/azul/zulu-openjdk:17@sha256:d1a6ac9a55017c199e81bd22d3953e1a17d8fcadd8bf259bdd78  0.0s
 => [rest builder 2/4] COPY . /app/                                                                                            0.1s
 => [rest builder 3/4] WORKDIR /app/                                                                                           0.1s
 => ERROR [rest builder 4/4] RUN ["./gradlew", "build", "shadowJar"]                                                           0.3s
------                                                                                                                              
 > [rest builder 4/4] RUN ["./gradlew", "build", "shadowJar"]:                                                                      
0.200 Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain
0.200 Caused by: java.lang.ClassNotFoundException: org.gradle.wrapper.GradleWrapperMain
------
failed to solve: process "./gradlew build shadowJar" did not complete successfully: exit code: 1
```

## Expected Behavior
The `docker compose up --build` command should complete successfully, as it does when checking out to commit `6bdf67708238757542cb5e2094c50fa4e2cacf44`.

--- a/.gitignore
+++ b/.gitignore
@@ -16,6 +16,7 @@
 *.gz
 *.iso
 *.jar
+!gradle/wrapper/gradle-wrapper.jar
 *.rar
 *.tar
 *.zip

--- a/iceberg-rest-catalog/.gitignore
+++ b/iceberg-rest-catalog/.gitignore
@@ -18,6 +18,3 @@ out
 build
 dependencies.lock
 **/dependencies.lock
-gradle/
-gradle/wrapper/gradle-wrapper.jar
-

--- a/iceberg-rest-catalog/gradle/wrapper/gradle-wrapper.properties
+++ b/iceberg-rest-catalog/gradle/wrapper/gradle-wrapper.properties
@@ -1,0 +1,7 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.10.2-bin.zip
+networkTimeout=10000
+validateDistributionUrl=true
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists

